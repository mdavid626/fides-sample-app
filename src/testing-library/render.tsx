import React from 'react';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import { render } from '@testing-library/react';

export const renderWithRouter = (component: JSX.Element) => {
  const router = {} as { current?: ReturnType<typeof createBrowserRouter> };
  return {
    ...render(component, {
      wrapper: ({ children }) => {
        const browserRouter = createBrowserRouter([
          {
            path: '',
            element: children,
          },
        ]);
        router.current = browserRouter;
        return <RouterProvider router={browserRouter} />;
      },
    }),
    router,
  };
};
