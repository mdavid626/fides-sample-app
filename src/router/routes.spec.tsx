import React from 'react';
import { cleanup, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { renderWithRouter } from '../testing-library/render';
import MoviesPage from '../pages/movies-page/movies-page';
import AboutPage from '../pages/about-page/about-page';
import FavouritesPage from '../pages/favourites-page/favourites-page';
import Routes from './routes';

jest.mock('../pages/movies-page/movies-page');
jest.mock('../pages/about-page/about-page');
jest.mock('../pages/favourites-page/favourites-page');

describe('routes', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockReturnValue(undefined);
    (MoviesPage as jest.Mock).mockReturnValue(<div>movies-page</div>);
    (AboutPage as jest.Mock).mockReturnValue(<div>about-page</div>);
    (FavouritesPage as jest.Mock).mockReturnValue(<div>favourites-page</div>);
  });
  afterEach(cleanup);
  afterEach(jest.resetAllMocks);

  const pages: string[] = ['/', '/favourites', '/about'];
  it.each(pages)('should render "%s" page', (page) => {
    const { asFragment } = renderWithRouter(<Routes />, undefined, [page]);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render default page when unknown route', () => {
    renderWithRouter(<Routes />, undefined, ['/unknown']);
    expect(screen.getByText('movies-page')).toBeVisible();
  });

  it('should throw error on error route', () => {
    expect(() => renderWithRouter(<Routes />, undefined, ['/error'])).toThrow(
      'test error'
    );
  });
});
