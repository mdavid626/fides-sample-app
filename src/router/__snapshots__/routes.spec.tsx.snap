// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes should render "/" page 1`] = `
<DocumentFragment>
  <div>
    movies-page
  </div>
</DocumentFragment>
`;

exports[`routes should render "/about" page 1`] = `
<DocumentFragment>
  <div>
    about-page
  </div>
</DocumentFragment>
`;

exports[`routes should render "/favourites" page 1`] = `
<DocumentFragment>
  <div>
    favourites-page
  </div>
</DocumentFragment>
`;
